src/integrated/pond_soil_model.jl

```diff
@@ -253,10 +259,11 @@ stored in `p.soil_infiltration`).
 function ClimaLSM.boundary_flux(
     bc::RunoffBC,
     ::TopBoundary,
+    model::Soil.RichardsModel,
```

Comment:
The signature for `boundary_flux` for soil models uses the `model`, e.g.:https://github.com/CliMA/ClimaLSM.jl/blob/f667b01789714e77cc3ad91e1fe7c3662ef67f44/src/standalone/Soil/rre.jl#L103

I think not passing `model` was a bug, but since this is mainly an example in a tutorial explaining how to set up LSMs, im not sure we ever really run a simulation and check the output. so i think this was a case of insufficient testing.

Reply:
will update docs & add type enforcement in PR417